generator.go

```diff
@@ -0,0 +1,4 @@
+//go:generate go run pkg/codegen/generator/cleanup/main.go
```

Comment:
I don't have a strong opinion; should this file be in the root? An alternative might be moving this to `pkg/codegen` but that might cause us to nest the current codegen dir one layer more

Reply:
It needs to be in root, without it in root, the controllers and clients created end up in weird places.